lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -46,17 +46,23 @@ def perform_request(http, env)
       end
 
       def configure_ssl(http, ssl)
-        http.verify_mode  = ssl_verify_mode(ssl)
-        http.cert_store   = ssl_cert_store(ssl)
+        http_set(http, :verify_mode, ssl_verify_mode(ssl))
+        http_set(http, :cert_store,  ssl_cert_store(ssl))
 
-        http.certificate  = ssl[:client_cert]  if ssl[:client_cert]
-        http.private_key  = ssl[:client_key]   if ssl[:client_key]
-        http.ca_file      = ssl[:ca_file]      if ssl[:ca_file]
-        http.ssl_version  = ssl[:version]      if ssl[:version]
+        http_set(http, :certificate, ssl[:client_cert]) if ssl[:client_cert]
+        http_set(http, :private_key, ssl[:client_key])  if ssl[:client_key]
+        http_set(http, :ca_file,     ssl[:ca_file])     if ssl[:ca_file]
+        http_set(http, :ssl_version, ssl[:version])     if ssl[:version]
       end
 
-      def cached_connection(url, proxy_uri)
-        (@cached_connection ||= {})[[url.scheme, url.host, url.port, proxy_uri]] ||= yield
+      def http_set(http, attr, value)
+        if http.sent(attr) != value
+          http.send("#{attr}=", value)
+        end
+      end
+
+      def cached_connection
+        @cached_connection ||= yield
```

Comment:
Removing the hash caching, we're going to reuse the same connection even though the url changes. Or am I missing something?

Reply:
`Net::HTTP::Persistent` has logic for using different connections for different URI bases (including differences in proxy) internally.

Reply:
in 2.9.4 https://github.com/drbrain/net-http-persistent/blob/v2.9.4/lib/net/http/persistent.rb#L613-L619
in 3.0.0 https://github.com/drbrain/net-http-persistent/blob/v3.0.0/lib/net/http/persistent.rb#L608-L611